src/Query/Terms.php

```diff
@@ -15,97 +15,90 @@
 class Terms extends AbstractQuery
 {
     /**
-     * Terms.
-     *
-     * @var array Terms
+     * @var string
      */
-    protected $_terms;
+    private $field;
 
     /**
-     * Terms key.
-     *
-     * @var string Terms key
+     * @var string[]
      */
-    protected $_key;
+    private $terms;
 
     /**
-     * Construct terms query.
-     *
-     * @param string $key   Terms key
-     * @param array  $terms Terms list
+     * @var string[]|null
+     */
+    private $lookup;
+
+    /**
+     * @param string[] $terms Terms list, leave empty if building a terms-lookup query
      */
-    public function __construct(string $key = '', array $terms = [])
+    public function __construct(string $field, array $terms = [])
     {
-        $this->setTerms($key, $terms);
+        if (empty($field)) {
+            throw new InvalidException('Terms field name has to be set');
+        }
+
+        $this->field = $field;
+        $this->terms = $terms;
     }
 
     /**
-     * Sets key and terms for the query.
-     *
-     * @param string $key   terms key
-     * @param array  $terms terms for the query
-     *
-     * @return $this
+     * Commodity function to build a Terms query, preconfigured for terms lookup.
      */
-    public function setTerms(string $key, array $terms): self
+    public static function buildTermsLookup(string $field, string $index, string $id, string $path): self
```

Comment:
Do we need this?

Reply:
I would say: yes :)
It is an helper function to directly create a Terms lookup query, without first creating the Terms query, and then assign the lookup settings.

WDYT?

Reply:
I wonder if the constructor should then allow this? ;-)

But I'm fine going with it.

Reply:
I'd say that the constructor covers the "base" case, while a specific method can configure the "lookup" settings.
Having an helper doing it in one single go, helps in this direction :)

Saying it here as, if we agree, this could be the standard for other query classes too. wdyt?

Reply:
Isn't this the same as:

```
$terms = (new Terms($field))->setTermsLookup($index, $id, $path);
```

I did not test if the above works.

What I worry is that we add methods which do not necessarily simplify the code on the user end.

Reply:
Yes, the above is the same of `Terms::buildTermsLookup($field, $index, $id, $path)`

Reply:
Do you think it is still needed? Or could we put the above for example as a doc comment for the constructor?

Reply:
I agree, I removed the helper function.
I guess looking at the Terms methods it is clear how to use the `setTermsLookup()` :)

See: #1767